lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
> So you are suggesting do not use GC.Collect

Yes, I think we shouldn't call the garbage collector. If that creates short-term pain, then we could do the configurable call, with the default being off.

The problem is much better addressed in the container layer, such that the CLR figures out when to call the GC. In all likelihood, it will make better decisions than us :-)